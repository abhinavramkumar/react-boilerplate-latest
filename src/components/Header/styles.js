import styled from "styled-components";
import { Link } from "react-router-dom";

export const Wrapper = styled.div`
  height: 40px;
  width: 100%;
  margin: 0;
  padding: 0;
`;

export const Container = styled.div`
  width: 90%;
  height: 100%;
  margin: 0 auto;
  padding: 0;
`;

export const Row = styled.div`
  height: 100%;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const Brand = styled.div`
  height: 100%;
  font-family: ${props => props.theme.fonts.sans};
  font-size: 26px;
  font-weight: 700;
  line-height: 40px;
  color: ${props => props.theme.colors.greyDark};
  word-spacing: 2px;
`;

export const Links = styled.div`
  display: flex;
  justify-content: flex-end;
  align-items: center;
  height: 100%;
`;

export const StyledLink = styled(Link)`
  margin: 0;
  padding: 0.5rem 1.25rem;
  font-size: 1rem;
  display: block;
  box-sizing: border-box;
  border: none;
  background: none;
  cursor: pointer;
  color: ${props => props.theme.colors.greyDark};
  text-decoration: none;
`;
