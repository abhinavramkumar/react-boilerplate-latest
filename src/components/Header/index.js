import React, { Component } from "react";
import { Wrapper, Container, Row, Brand, Button } from "./styles.js";

class Header extends Component {
  state = {};

  render() {
    return (
      <Wrapper>
        <Container>
          <Row>
            <Brand>Brand</Brand>
            <Button>Logout</Button>
          </Row>
        </Container>
      </Wrapper>
    );
  }
}

export default Header;
