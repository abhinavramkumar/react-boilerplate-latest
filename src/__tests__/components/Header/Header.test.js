import React from "react";
import { shallow } from "enzyme";
import Header from "../../../components/Header";
import "jest-styled-components";

test("should render Header component", () => {
  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();
});
