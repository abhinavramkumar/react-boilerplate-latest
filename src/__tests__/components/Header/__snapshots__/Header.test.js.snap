// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header component 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <styled.div>
        React Boilerplate
      </styled.div>
      <styled.div>
        <Styled(Link)
          to="/"
        >
          Home
        </Styled(Link)>
        <Styled(Link)
          to="/login"
        >
          Login
        </Styled(Link)>
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;
